src/components/Button/Button.vue

```diff
@@ -0,0 +1,294 @@
+			},
+			default: '',
+		},
+
+		/**
+		 * Specifies whether the button should span all the available width.
+		 * By default, buttons span the whole width of the container.
+		 */
+		wide: {
+			type: Boolean,
+			default: true,
+		},
+	},
+
+	data() {
+		return {
+			/**
+			 * Keeps track of whether the element's focus status is due to having
+			 * tabbed to it. We use this to display a thick 'focus outline' only
+			 * when the user is navigating with the keyboard.
+			 */
+			tabbed: false,
+		}
+	},
+
+	computed: {
+		hasText() {
+			return this.text
+		},
+
+		hasIcon() {
+			return this.$slots.default !== undefined
+		},
+
+		iconOnly() {
+			return this.hasIcon && !this.hasText
+		},
+
+		textOnly() {
+			return !this.hasIcon && this.hasText
+		},
+
+		iconAndText() {
+			return this.hasIcon && this.hasText
+		},
+
+		// Classes applied to the button element
+		buttonClassObject() {
+			return {
+				// If icon only, some additional css rules are required
+				'button-vue--icon-only': this.iconOnly,
+				'button-vue--text-only': this.textOnly,
+				'button-vue--icon-and-text': this.iconAndText,
+				'vue-default': !this.color,
+				[`vue-${this.color}`]: this.color,
+				wide: this.wide,
+				tabbed: this.tabbed,
+			}
+		},
+	},
+
+	methods: {
+		// Forward the click event to the parent component
+		handleClick(event) {
+			this.$emit('click', event)
+		},
+
+		// When the tab key is lifted, the button has been "tabbed in"
+		handleTabUp() {
+			this.tabbed = true
+		},
+
+		// Everytime the button is blurred, we remove the tabbed state
+		handleBlur() {
+			this.tabbed = false
+		},
+	},
+}
+
+</script>
+
+<style lang="scss" scoped>
+
+.button-vue {
+	position: relative;
+	overflow: hidden;
+	border: 0;
+	font-size: var(--default-font-size);
+	font-weight: bold;
+	height: $clickable-area;
+	min-width: $clickable-area;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	border-radius: var(--border-radius-pill);
+	transition: background-color 0.1s linear !important;
+	transition: border 0.1s linear;
+
+	// No outline feedback for focus. Handled with a toggled class in js (see data)
+	&:focus {
+		outline: none;
+	}
+
+	// No active state for buttons
+	&:active {
+		background-color: unset;
+	}
+
+	&__wrapper {
+		display: inline-flex;
+		align-items: center;
+		justify-content: space-around;
+	}
+
+	&__icon {
+		height: 44px;
+		width: 44px;
+		display: flex;
+		justify-content: center;
+		align-items: center;
+	}
+	&__text {
+		font-weight: bold;
+	}
+
+	// Icon-only button
+	&--icon-only {
+		width: $clickable-area !important;
+	}
+
+	// Text-only button
+	&--text-only {
+		padding: 0 12px;
+	}
+
+	// Icon and text button
+	&--icon-and-text {
+		padding: 0 16px 0 4px;
+	}
+
+	// Wide button spans the whole width of the container
+	&.wide {
+		width: 100%;
+	}
+
+	// We use around our buttons instead of an outline, so that the added "border"
+	// coincides with the border of the element.
```

Comment:
Is the outline not ok?

Reply:
No, because it's not possible to add a border radius to it, it's always a rectangle around the element and it doesn't look good with our buttons

Reply:
To be checked about the usual process on accessibility. It might be a bad thing to round the outline. I'm lacking knowledge here :)

Reply:
Checked and it's acceptable and widely adopted :)